util/git/client.go

```diff
@@ -80,11 +80,23 @@ type Client interface {
 	IsAnnotatedTag(string) bool
 	ChangedFiles(revision string, targetRevision string) ([]string, error)
 	IsRevisionPresent(revision string) bool
+	// SetAuthor sets the author name and email in the git configuration.
+	SetAuthor(name, email string) (string, error)
+	// CheckoutOrOrphan checks out the sync branch. If the branch does not exist, it creates an orphan branch.
```

Comment:
You are documenting with as "sync branch" but this function can be used in different contexts than hydrator.